1.2/percentile_test.go

```diff
@@ -6,53 +6,58 @@ import (
 )
 
 func TestGeneratePercentile(t *testing.T) {
+
 	var tests = []struct {
-		fn percentile
-		p  uint
+		input float32
+		p     uint
 	}{
-		{generatePercentile(.05), 15},
-		{generatePercentile(.3), 20},
-		{generatePercentile(.4), 20},
-		{generatePercentile(.5), 35},
-		{generatePercentile(1.0), 50},
+		{.05, 0},
+		{.3, 15},
+		{.4, 20},
+		{.5, 20},
+		{1.0, 50},
 	}
-	data := []uint{15, 20, 35, 40, 50}
+	data := []uint{15, 40, 35, 20, 50, 0}
+
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("PASS"), func(t *testing.T) {
-			if output, err := test.fn(data); output != test.p || err != nil {
+			if output, err := generatePercentile(test.input); err != nil {
+				t.Errorf("Got error, exprected no error")
```

Comment:
This throws away the information from `err`. Can you include that in the test output?